lib/src/controls/flyouts/flyout.dart

```diff
@@ -0,0 +1,779 @@
+  ///
+  /// If [placementMode] is auto, [autoModeConfiguration] is taken in consideration
+  /// to determine the correct placement mode
+  ///
+  /// [forceAvailableSpace] determines whether the flyout size should be forced
+  /// the available space according to the attached target. It's useful when the
+  /// flyout is large but can not be on top of the target. Defaults to false
+  ///
+  /// [shouldConstrainToRootBounds], when true, the flyout is limited to the
+  /// bounds of the closest [Navigator]. If false, the flyout may overflow the
+  /// screen on all sides. Defaults to `true`
+  ///
+  /// [additionalOffset] is the offset of the flyout around the attached target
+  ///
+  /// [margin] is the margin of the flyout to the root bounds
+  ///
+  /// If there isn't a [Navigator] in the tree, a [navigatorKey] can be used to
+  /// display the flyout. If null, [Navigator.of] is used.
+  ///
+  /// [transitionBuilder] builds the transition. By default, a slide-fade transition
+  /// is used on vertical directions; and a fade transition in horizontal directions.
+  /// The default fade animation can not be disabled.
+  ///
+  /// [transitionDuration] configures the duration of the transition animation.
+  /// By default, [ThemeData.fastAnimationDuration] is used. Set to [Duration.zero]
+  /// to disable transitions at all
+  ///
+  /// [position] lets you position the flyout anywhere on the screen, making it
+  /// possible to create context menus. If provided, [placementMode] is ignored.
+  Future<T?> showFlyout<T>({
+    required WidgetBuilder builder,
+    bool barrierDismissible = true,
+    bool dismissWithEsc = true,
+    bool dismissOnPointerMoveAway = false,
+    FlyoutPlacementMode placementMode = FlyoutPlacementMode.auto,
+    FlyoutAutoConfiguration? autoModeConfiguration,
+    bool forceAvailableSpace = false,
+    bool shouldConstrainToRootBounds = true,
+    double additionalOffset = 8.0,
+    double margin = 8.0,
+    Color? barrierColor,
+    NavigatorState? navigatorKey,
+    FlyoutTransitionBuilder? transitionBuilder,
+    Duration? transitionDuration,
+    Offset? position,
+  }) async {
+    _ensureAttached();
+    assert(_attachState!.mounted);
+
+    final context = _attachState!.context;
+    assert(debugCheckHasFluentTheme(context));
+
+    final theme = FluentTheme.of(context);
+    transitionDuration ??= theme.fastAnimationDuration;
+
+    final navigator = navigatorKey ?? Navigator.of(context);
+    final navigatorBox = navigator.context.findRenderObject() as RenderBox;
+
+    final targetBox = context.findRenderObject() as RenderBox;
+    final targetSize = targetBox.size;
+    final targetOffset =
+        targetBox.localToGlobal(Offset.zero, ancestor: navigatorBox) +
+            Offset(0, targetSize.height);
+    final targetRect =
+        targetBox.localToGlobal(Offset.zero, ancestor: navigatorBox) &
+            targetSize;
+
+    _open = true;
+    notifyListeners();
+
+    final flyoutKey = GlobalKey();
```

Comment:
I don't know if having a global key live in this scope is safe, can you confirm?

Reply:
yeah, it is safe because we push a new route.